pvlib/horizon.py

```diff
@@ -0,0 +1,646 @@
+
+    Returns
+    -------
+    points: Nx3 ndarray
+        Array with N (lat, lon, elev) points that lie on the surface of the
+        triangle.
+
+    [1] http://graphics.stanford.edu/courses/cs468-08-fall/pdf/osada.pdf
+    """
+    c1 = tools.lle_to_xyz(p1)
+    c2 = tools.lle_to_xyz(p2)
+    c3 = tools.lle_to_xyz(p3)
+
+    r1 = np.random.rand(num_samples).reshape((num_samples, 1))
+    r2 = np.random.rand(num_samples).reshape((num_samples, 1))
+    sqrt_r1 = np.sqrt(r1)
+
+    random_pts = (1-sqrt_r1)*c1 + sqrt_r1*(1-r2)*c2 + sqrt_r1*r2*c3
+    random_pts = tools.xyz_to_lle(random_pts)
+
+    return random_pts
+
+
+def filter_points(horizon_azimuths, horizon_angles, bin_size=1):
+    """
+    Bins the horizon_points by azimuth values. The azimuth value of each
+    point in horizon_points is rounded to the nearest bin and then the
+    max value in each bin is returned.
+
+    Parameters
+    ----------
+    horizon_azimuths: numeric
+        Azimuth values for points that define the horizon profile. The ith
+        element in this array corresponds to the ith element in horizon_angles.
+
+    horizon_angles: numeric
+        Elevation angle values for points that define the horizon profile. The
+        elevation angle of the horizon is the angle that the horizon makes with
+        the horizontal. It is given in degrees above the horizontal. The ith
+        element in this array corresponds to the ith element in
+        horizon_azimuths.
+
+    bin_size : int
+        The width of the bins for the azimuth values.
+
+    Returns
+    -------
+    filtered_azimuths: numeric
+        Azimuth values for points that define the horizon profile. The ith
+        element in this array corresponds to the ith element in
+        filtered_angles.
+
+    filtered_angles: numeric
+        elevation angle values for points that define the horizon profile given
+        in degrees above the horizontal. The ith element in this array
+        corresponds to the ith element in filtered_azimuths.
+
+    """
+    assert(horizon_azimuths.shape[0] == horizon_angles.shape[0])
+
+    wedges = {}
+    for i in range(horizon_angles.shape[0]):
+        azimuth = horizon_azimuths[i]
+        elevation = horizon_angles[i]
+        azimuth_wedge = tools.round_to_nearest(azimuth, bin_size)
+
+        if azimuth_wedge in wedges:
+            wedges[azimuth_wedge] = max(elevation, wedges[azimuth_wedge])
+        else:
+            wedges[azimuth_wedge] = elevation
+
+    filtered_azimuths = []
+    filtered_angles = []
+    for key in sorted(wedges.keys()):
+        filtered_azimuths.append(key)
+        filtered_angles.append(wedges[key])
+
+    filtered_angles = np.array(filtered_angles)
+    filtered_azimuths = np.array(filtered_azimuths)
+    return filtered_azimuths, filtered_angles
+
+
+def collection_plane_elev_angle(surface_tilt, surface_azimuth, direction):
+    """
+    Determine the elevation angle created by the surface of a tilted plane
```

Comment:
The description is unclear to me. I think this function is calculating the elevation angle of the unit normal after rotation around the earth surface normal to a different azimuth `direction`, ?